### ISSUE TITLE
Cannot use filter contains 

### ISSUE THREAD

**Issue by mmedinam1600** [chapter-three/next-drupal#369]
I am following the tutorial using SearchAPI with solr.
https://next-drupal.org/docs/reference/getsearch

I am using the following function via the next api:

```js
await getSearchIndex(index, body);
```

Everything works fine for me, except when I want to add a filter to the search with the 'CONTAINS' operator.

I am using [JSONAPIParams](https://www.npmjs.com/package/drupal-jsonapi-params) to build the following filter:

```js
params.addGroup("or", "OR");
params.addFilter('title', payload.value, 'CONTAINS', 'or');
```

In drupal I have an index that contains the `title` field of type `string`.
![image](https://user-images.githubusercontent.com/29582013/204058296-a9b03e3a-dddf-4415-8801-bd7948c7ea8c.png)

And when I search with this filter, I get the following error:

![image](https://user-images.githubusercontent.com/29582013/204057268-12a4c966-ec9c-4366-bbfa-43b68de1164b.png)

```
Location: http://tcmc-web.lndo.site/jsonapi/index/index?fields%5Bnode--inmueble%5D=body%2Ctitle%2Cpath%2Cuid%2Ccreated%2Cfield_imagenes%2Cfield_link_imagen%2Cfield_property_type_id%2Cfield_toilet_number%2Cfield_bedrooms%2Cfield_parkingspaces%2Cfield_price%2Cfield_currency_iso%2Cfield_municipality%2Cfield_state%2Cfield_street%2Cfield_ubicacion%2Cfield_restrooms&filter%5Band%5D%5Bgroup%5D%5Bconjunction%5D=AND&filter%5Bor%5D%5Bgroup%5D%5Bconjunction%5D=OR&filter%5Btitle%5D%5Bcondition%5D%5Bpath%5D=title&filter%5Btitle%5D%5Bcondition%5D%5Bvalue%5D=test&filter%5Btitle%5D%5Bcondition%5D%5Boperator%5D=CONTAINS&filter%5Btitle%5D%5Bcondition%5D%5BmemberOf%5D=or&include=field_imagenes.field_media_image%2Cuid%2Cfield_property_type_id%2Cfield_agencia%2Cfield_agent_id%2Cfield_municipality%2Cfield_state&page%5Blimit%5D=20&page%5Boffset%5D=0
Message: Drupal\Core\Http\Exception\CacheableBadRequestHttpException: The 'CONTAINS' operator is not allowed in a filter parameter. in Drupal\jsonapi_search_api\Resource\IndexResource->applyFiltersToQuery() (line 164 of /app/web/modules/contrib/jsonapi_search_api/src/Resource/IndexResource.php).
```

I have not been able to fix this error yet, it seems to me that it has to do with Drupal, but I don't know how to fix it.

In short, what I need is the following:

I have some nodes with title:
```
* Dallas home with pool <- Node
* 3 story house in Irving <- Node
* Concrete house in Dallas <- Node
```

In a text input, I enter
`Dallas`

And the answer I'm waiting for is: (2 results)
```
* **Dallas** home with pool <- Node
* Concrete house in **Dallas** <- Node
```

I hope you can help me and if you need any other information, do not hesitate to ask me.

**Comment by shadcn** [comment]
I tried to reproduce this and can confirm I can see this too. I'll look into it and report back.

**Comment by shadcn** [comment]
OK found it. This is from the `jsonapi_search_api` module.

```php
/**
 * A condition object for the EntityQuery.
 */
class EntityCondition extends BaseEntityCondition {

  /**
   * The allowed condition operators.
   *
   * The operators 'STARTS_WITH', 'CONTAINS', 'ENDS_WITH' are not supported by <------- 👈
   * the Search API module. <------- 👈
   *
   * @var string[]
   *
   * @see \Drupal\search_api\Query\ConditionSetInterface
   */
  public static $allowedOperators = [
    '=', '<>',
    '>', '>=', '<', '<=',
    'IN', 'NOT IN',
    'BETWEEN', 'NOT BETWEEN',
    'IS NULL', 'IS NOT NULL',
  ];

}
```

Let me see how we can implement fuzzy search.

**Comment by shadcn** [comment]
You can use **Fulltext** field type for your index. This will fuzzy match all titles containing your keyword.

![kj2BnhAu](https://user-images.githubusercontent.com/124599/204322719-5060dd16-d28d-462b-b4c8-80679304a059.png)

```ts
// Filter all title with the word dallas
const params = new DrupalJsonApiParams().addFilter("title", "dallas")

const nodes = await drupal.getSearchIndexFromContext("content", context, {
  params: params.getQueryObject(),
})
```

**Comment by mmedinam1600** [comment]
Thank you very much, I will try the proposed solution. I have been trying for days to come up with a solution.

**Comment by mmedinam1600** [comment]
It worked perfectly, in case it is useful to someone else: A facet can not be created with the Fulltext type, just have the field you want to filter in the index is enough.
